tasks.py

```diff
@@ -294,7 +294,7 @@ def validate_kind_version():
 
 def generate_manifest(ctx, crd_options="crd:crdVersions=v1", bgp_type="native", output=None, with_prometheus=False):
     _fetch_kubectl()
-    run("GOPATH={} go install sigs.k8s.io/controller-tools/cmd/controller-gen@{}".format(build_path, controller_gen_version))
+    run("GOBIN={}/bin/ GOPATH={} go install sigs.k8s.io/controller-tools/cmd/controller-gen@{}".format(build_path, build_path, controller_gen_version))
```

Comment:
I think if we override GOBIN, GOPATH becomes irrelevant.

Reply:
:santa:

Reply:
When running this on my laptop with the following set (I admit that I may have a very unorthodox setup):
```
[akaris@workstation metallb (issue2218)]$ env | grep GO
GOBIN=/home/akaris/development/go/bin
GOPATH=/home/akaris/development/go
```

I run into:
```
No resources found
go: sigs.k8s.io/controller-tools/cmd/controller-gen@/home/akaris/development/metallb/build: sigs.k8s.io/controller-tools/cmd/controller-gen@/home/akaris/development/metallb/build: invalid version: unknown revision /home/akaris/development/metallb/build
```

when I unset GOPATH.

That said, I'll push another small modification and with the 2 run() commands in generate_manifest verbose, because in case of a failure, it's difficult to identify which command actually failed:
```
GOBIN=/home/akaris/development/metallb/build/bin/ go install sigs.k8s.io/controller-tools/cmd/controller-gen@/home/akaris/development/metallb/build
go: sigs.k8s.io/controller-tools/cmd/controller-gen@/home/akaris/development/metallb/build: sigs.k8s.io/controller-tools/cmd/controller-gen@/home/akaris/development/metallb/build: invalid version: unknown revision /home/akaris/development/metallb/build
```

Reply:
So now, it looks like this:
```
/home/akaris/development/metallb/build/kubectl delete po -n metallb-system --all
No resources found
GOBIN=/home/akaris/development/metallb/build/bin/ GOPATH=/home/akaris/development/metallb/build go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.11.1
go: downloading sigs.k8s.io/controller-tools v0.11.1
go: downloading k8s.io/apimachinery v0.26.0
go: downloading golang.org/x/tools v0.4.0
go: downloading k8s.io/apiextensions-apiserver v0.26.0
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading gopkg.in/yaml.v3 v3.0.1
go: downloading k8s.io/api v0.26.0
go: downloading github.com/spf13/cobra v1.6.1
go: downloading sigs.k8s.io/yaml v1.3.0
go: downloading github.com/gobuffalo/flect v0.3.0
go: downloading github.com/fatih/color v1.13.0
go: downloading github.com/mattn/go-isatty v0.0.14
go: downloading github.com/mattn/go-colorable v0.1.9
go: downloading github.com/spf13/pflag v1.0.5
go: downloading golang.org/x/sys v0.3.0
go: downloading golang.org/x/mod v0.7.0
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading k8s.io/utils v0.0.0-20221107191617-1a15be271d1d
go: downloading github.com/google/gofuzz v1.1.0
go: downloading k8s.io/klog/v2 v2.80.1
go: downloading gopkg.in/inf.v0 v0.9.1
go: downloading golang.org/x/net v0.4.0
go: downloading sigs.k8s.io/structured-merge-diff/v4 v4.2.3
go: downloading sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2
go: downloading golang.org/x/text v0.5.0
go: downloading github.com/go-logr/logr v1.2.3
go: downloading github.com/json-iterator/go v1.1.12
go: downloading github.com/modern-go/reflect2 v1.0.2
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
/home/akaris/development/metallb/build/bin/controller-gen crd:crdVersions=v1 rbac:roleName=manager-role webhook paths="./api/..." output:crd:artifacts:config=config/crd/bases
# Warning: 'patchesStrategicMerge' is deprecated. Please use 'patches' instead. Run 'kustomize edit fix' to update your Kustomization automatically.
/home/akaris/development/metallb/build/kubectl apply -f /tmp/tmpzvcxkvkr/metallb.yaml
```